components/fxa-client/ffi/src/lib.rs

```diff
@@ -591,3 +591,10 @@ pub extern "C" fn fxa_send_tab(
 define_handle_map_deleter!(ACCOUNTS, fxa_free);
 define_string_destructor!(fxa_str_free);
 define_bytebuffer_destructor!(fxa_bytebuffer_free);
+
+/// Gather telemetry collected by FxA.
+#[no_mangle]
+pub extern "C" fn fxa_gather_telemetry(handle: u64, error: &mut ExternError) -> *mut c_char {
```

Comment:
Returning a JSON bundle here seems OK to be, for the b/w-compat reasons you describe. It's kinda weird but if we do a good job of documenting it and setting expectations that this API will eventually go away in favour of glean, then 👍